lib/src/control_panel/number_stepper_control.dart

```diff
@@ -25,7 +26,17 @@ class NumberStepperControl extends StatelessWidget {
   /// Executed when the decrement button is pressed.
   final VoidCallback onIncrementPressed;
 
-  String get _displayTextScale => value.toStringAsFixed(1);
+  /// Used to customize how the [value] is displayed in the
+  /// [EnvironmentControlPanel].
+  final String Function(num)? valueDisplayBuilder;
```

Comment:
At this point, is it simpler to just declare "value" as a string? You could have multiple factories for NumberStepperControl, like NumberStepperControl.withStringValue(title, someString, onDecrement, onIncrement) etc.

Reply:
My initial answer was going to be "no", but the more I think about this idea, the more I like it. Making this control a general stepper (instead of specifically a number stepper) would allow this to be used in new ways (adjusting a log level enum, for example). Sold! :)

Reply:
And done.

Reply:
Nice! It doesn't even add application code in this case, because the source of truth is still elsewhere :-D